src/utils/hooks/useFetchUserQuery.ts

```diff
@@ -0,0 +1,17 @@
+import { useQuery } from '@tanstack/react-query';
+import { useTranslation } from 'react-i18next';
+import { fetchUser } from '../../api/roleApi';
+
+export const useFetchUserQuery = (username: string) => {
+  const { t } = useTranslation();
+
+  return useQuery({
+    enabled: !!username,
+    queryKey: ['user', username],
+    queryFn: () => fetchUser(username),
+    meta: { errorMessage: t('feedback.error.get_person') },
+    retry: 0,
+    staleTime: Infinity,
+    gcTime: 1_800_000, // 30 minutes
+  });
+};
```

Comment:
I suggest placing this in /api/hooks since it's an API query

Reply:
Good catch. Will replace it in another PR once the /api/hooks folder is created by [5683](https://github.com/BIBSYSDEV/NVA-Frontend/pull/5683) 👍